crates/daemon/src/rpc_server.rs

```diff
@@ -601,7 +595,8 @@ impl RpcServer {
             SYSTEM_OBJECT,
             session.clone(),
         ) {
-            if let Ok(value) = self.clone().watch_command_task(task_id) {
+            let task_id = task_handle.task_id();
+            if let Ok(value) = self.clone().watch_command_task(task_handle) {
```

Comment:
Nit: clone `task_id` here, and use it on line 630, instead of re-fetching from `task_handle`?

Reply:
Awgh, is awkward because this is down in a deeper lexical scope.

Limited need for gymnastics considering the compiler will just inline it anyways.

Reply:
Wait, actually these are two different tasks.  

I'll clarify by renaming

Reply:
Cool, sorry for the noise then and thanks for making it clearer